sdk/python/feast/infra/offline_stores/contrib/trino_offline_store/trino_queries.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+import datetime
+import os
+import signal
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import trino
+from trino.dbapi import Cursor
+from trino.exceptions import TrinoQueryError
+
+from feast.infra.offline_stores.contrib.trino_offline_store.trino_type_map import (
+    trino_to_pa_value_type,
+)
+
+
+class QueryStatus(Enum):
+    PENDING = 0
+    RUNNING = 1
+    ERROR = 2
+    COMPLETED = 3
+    CANCELLED = 4
+
+
+class Trino:
+    def __init__(
+        self,
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        user: Optional[str] = None,
+        catalog: Optional[str] = None,
+        auth: Optional[Any] = None,
+        http_scheme: Optional[str] = None,
+    ):
+        self.host = host or os.getenv("TRINO_HOST")
```

Comment:
Nit, we should assert that all th necessary values are set at the end of the constructor or raise an error

Reply:
good idea. I added that validation